R/utils.R

```diff
@@ -50,3 +50,37 @@ remove_as_is_class <- function(x) {
 diff_to_trunc <- function(x) {
   x - trunc(x)
 }
+
+# Only check if we have color support once per session
+has_color <- local({
+  has_color <- NULL
+  function(forget = FALSE) {
+    if (is.null(has_color) || forget) {
+      has_color <<- crayon::has_color()
+    }
+    has_color
+  }
+})
+
```

Comment:
The problem with caching this is that this will behave differently:
```
mtcarst <- tibble::as_tibble(mtcars)
print(mtcarst)
sink(tmp <- tempfile())
print(mtcarst)
sink(NULL)
```

I.e.  the color sequences will be also written to the sink. This might be OK, tough, but we just must be aware of it, maybe even document it.